docs/design/keyboard-shortcuts.md

```diff
@@ -1,314 +1,473 @@
+        - **XML manifest**: The value of the "actions.id" property in the shortcuts JSON file.
+    - The second parameter is the function that runs when a user presses the key combination that's mapped to an action.
 
     ```javascript
-    Office.actions.associate("SHOWTASKPANE", showTaskPane);
-    Office.actions.associate("HIDETASKPANE", hideTaskPane);
-
-    function showTaskPane() {
-        return Office.addin.showAsTaskpane()
-            .then(() => {
-                console.log("Task pane is visible.");
-            })
-            .catch((error) => {
-                console.log(error.code);
-            });
-    }
+    Office.actions.associate("ShowTaskpane", () => {
+        return Office.addin.showAsTaskpane()
+            .then(() => {
+                return;
+            })
+            .catch((error) => {
+                return error.code;
+            });
+    });
+    ```
 
-    function hideTaskPane() {
-        return Office.addin.hide()
-            .then(() => {
-                console.log("Task pane is hidden.");
-            })
-            .catch((error) => {
-                console.log(error.code);
-            })
-    }
+    ```javascript
+    Office.actions.associate("HideTaskpane", () => {
+        return Office.addin.hide()
+            .then(() => {
+                return;
+            })
+            .catch((error) => {
+                return error.code;
+            });
+    });
     ```
 
-Following the previous steps lets your add-in toggle the visibility of the task pane by pressing **Ctrl+Alt+Up** and **Ctrl+Alt+Down**. The same behavior is shown in the [Excel keyboard shortcuts](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Samples/excel-keyboard-shortcuts) sample in the Office Add-in Samples repo in GitHub.
+## Guidelines for custom key combinations
 
-## Details and restrictions
+Use the following guidelines to create custom key combinations for your add-ins.
 
-### Construct the action objects
+- A keyboard shortcut must include at least one modifier key (Alt/Option, Ctrl/Command, Shift) and only one other key. These keys must be joined with a `+` character.
+- The Command modifier key is supported on the macOS platform.
+- On macOS, the Alt key is mapped to the Option key. On Windows, the Command key is mapped to the Ctrl key.
+- The Shift key can't be used as the only modifier key. It must be combined with either Alt/Option or Ctrl/Command.
+- Key combinations can include characters "A-Z", "a-z", "0-9", and the punctuation marks "-", "_", and "+". By convention, lowercase letters aren't used in keyboard shortcuts.
+- When two characters are linked to the same physical key on a standard keyboard, then they're synonyms in a custom keyboard shortcut. For example, Alt+a and Alt+A are the same shortcut, as well as Ctrl+- and Ctrl+\_ ("-" and "_" are the linked to the same physical key).
```

Comment:
Is the `\` in `Ctrl+\_` a typo?

Reply:
No. It's needed to escape the underscore.